src/load_balancer/load_balancer.rs

```diff
@@ -76,30 +85,79 @@ impl LoadBalancer {
     }
 
     async fn get_worker(&mut self) -> Result<String, LoadBalancerError> {
-        // round-robin algo for selecting active worker
-        let cur_worker = self.current_worker;
-        let round_robin_sql = format!("(({cur_worker} - 1) % (select count(*) from {DF_TABLE_NAME})) + 1");
-        let df = self.ctx.sql(&format!("select id, worker_name, port_name from {DF_TABLE_NAME} where id = {round_robin_sql}"))
-            .await
-            .map_err(|e| LoadBalancerError::UnexpectedError(e.into()))?;
-        let workers = Worker::to_records(df)
-            .await
-            .map_err(|e| LoadBalancerError::UnexpectedError(e.into()))?;
-        let worker = workers.get(0)
-            .wrap_err("workers is empty")
-            .map_err(|e| LoadBalancerError::UnexpectedError(e))?;
-        let worker_name = worker.name.clone()
-            .wrap_err("worker name is empty")
-            .map_err(|e| LoadBalancerError::UnexpectedError(e))?;
-        let worker_port = worker.port.clone()
-            .wrap_err("worker port is empty")
-            .map_err(|e| LoadBalancerError::UnexpectedError(e))?;
-        let worker = format!("http://{}:{}", worker_name, worker_port);
-        if !validate_address(&worker)? {
-            return Err(LoadBalancerError::InvalidWorkerHostAddress);
-        }
-        self.current_worker += 1;
+        match self.algorithm {
+            Algorithm::RoundRobin => {
```

Comment:
I may be ahead of the game here - but could this codeblock be made into generic or trait objects? Seeing that we're returning Result<String, LoadBalancerError> for both of the arms?

Reply:
This is something I'll need to do in the future